server/pkg/integrationapi/item.go

```diff
@@ -28,12 +29,13 @@ func NewVersionedItem(ver item.Versioned) VersionedItem {
 	}
 }
 
-func NewItem(i *item.Item) Item {
+func NewItem(i *item.Item, s *schema.Schema) Item {
 	fs := lo.Map(i.Fields(), func(f *item.Field, _ int) Field {
+		sf := s.Field(f.FieldID())
```

Comment:
Note that `sf` can be nil. If nil, drop the current field (use `lo.FilterMap` instead of `lo.Map`)